dask/highlevelgraph.py

```diff
@@ -433,7 +444,12 @@ def _from_collection(cls, name, layer, collection):
         return cls(layers, deps)
 
     @classmethod
-    def from_collections(cls, name, layer, dependencies=()):
+    def from_collections(
+        cls,
+        name: str,
+        layer: Graph,
```

Comment:
See above.

Reply:
This is the signature of a public method; I'd rather not change it

Reply:
it may make it easier to understand if you think that a Layer object is a special case of a subgraph, and a subgraph is a special case of a graph

Reply:
Works for me, I'll treat this as food for thought and propose something better if I ever come up with it :)